packages/base/src/3dview/mainview.tsx

```diff
@@ -422,20 +422,68 @@ export class MainView extends React.Component<IProps, IStates> {
         const obj = this._model.sharedModel.getObjectByName(objectName);
 
         if (obj && obj.parameters && obj.parameters.Placement) {
-          const positionArray = obj?.parameters?.Placement?.Position;
-          const newPosition = [
-            positionArray[0] + updatedPosition.x,
-            positionArray[1] + updatedPosition.y,
-            positionArray[2] + updatedPosition.z
-          ];
-
-          this._mainViewModel.maybeUpdateObjectParameters(objectName, {
-            ...obj.parameters,
-            Placement: {
-              ...obj.parameters.Placement,
-              Position: newPosition
-            }
-          });
+          if (this._transformControls.mode === 'translate') {
+            const positionArray = obj?.parameters?.Placement?.Position;
+            const newPosition = [
+              positionArray[0] + updatedPosition.x,
+              positionArray[1] + updatedPosition.y,
+              positionArray[2] + updatedPosition.z
+            ];
+
+            this._mainViewModel.maybeUpdateObjectParameters(objectName, {
+              ...obj.parameters,
+              Placement: {
+                ...obj.parameters.Placement,
+                Position: newPosition
+              }
+            });
+          } else if (this._transformControls.mode === 'rotate' && this._pivot) {
+            // Retrieve the existing rotation from the shared model
+            const existingRotationArray = obj?.parameters?.Placement?.Axis;
+            const existingAngleDeg = obj?.parameters?.Placement?.Angle;
+            const existingAngleRad = THREE.MathUtils.degToRad(existingAngleDeg);
+            const existingAxis = new THREE.Vector3(
+              existingRotationArray[0],
+              existingRotationArray[1],
+              existingRotationArray[2]
+            ).normalize();
+            const existingQuaternion = new THREE.Quaternion().setFromAxisAngle(
+              existingAxis,
+              existingAngleRad
+            );
+
+            // Get the pivot rotation as a quaternion
+            const pivotQuaternion = new THREE.Quaternion().setFromEuler(
+              this._pivot.rotation
+            );
+
+            // Combine the existing and pivot rotations
+            const finalQuaternion =
+              existingQuaternion.multiply(pivotQuaternion);
+
+            const angle = 2 * Math.acos(finalQuaternion.w);
+            const angleDeg = THREE.MathUtils.radToDeg(angle);
+            const axis = new THREE.Vector3();
+            axis.set(finalQuaternion.x, finalQuaternion.y, finalQuaternion.z);
+
+            // Update the shared model with the new axis and angle
+            this._model.sharedModel.updateObjectByName(objectName, {
```

Comment:
This should be using the new `this._mainViewModel.maybeUpdateObjectParameters`

Reply:
for rotation the controls have been attaching to pivot only [see here](https://github.com/jupytercad/JupyterCAD/pull/574/files/e78d8c8954513076a5c529ee0ed9e7c6f2807dbb#diff-37a7f3f4d02e2bc4c4f40a612b75d24569d7c0c46420298e3281f4af6e29b7e6R1252)

For translation i've kept the logic straight and same as before